ast.ts

```diff
@@ -70,13 +72,15 @@ export function isAssignable(lhs: Type, rhs: Type): boolean {
 }
 
 export function isSubType(lhs: Type, rhs: Type) {
-    if (isClass(lhs)) {
-        return isTypeEqual(rhs, { tag: "none" });
-    } else if (lhs.tag === "list") {
+    if (lhs.tag === "list") {
         if (rhs.tag === "list")
             return isEmptyList(rhs) || isAssignable(lhs.type, rhs.type);
         else
             return isTypeEqual(rhs, { tag: "none" });
+    } else if (rhs.tag === "list") {
+        return isAssignable(lhs, rhs.type);
```

Comment:
what is this for? can you give an example?

Reply:
tc.ts:558 in master (tcStmt, case for, `if (!isAssignable(newCnt.a, newArray.a))`), if `newCnt` is a class object and `newArray` is a list of class objects, the previous code will only compare `newCnt.a` with `none` and exit, which is not right.

Reply:
I think you should compare `newCnt.a`, `newArray.a,type` instead of changing this function